contribs/gmf/src/controllers/AbstractDesktopController.js

```diff
@@ -263,28 +402,61 @@ export class AbstractDesktopController extends AbstractAPIController {
    * @private
    * @hidden
    */
-  setDataPanelMaxResizableWidth_() {
-    let rightPanelWidth = 320;
-    if (this.googleStreetViewActive) {
-      rightPanelWidth += 140;
-    }
-
+  setDataPanelMaxResizableWidth_(newToolsPanelWidth) {
     const minimumMapWidth = 120;
     const windowWidth = window.innerWidth;
+    // this.$toolsPanel_.width() often has not the new value yet after resizing, so take it from the resizing event!
+    const toolsPanelWidth = newToolsPanelWidth ? newToolsPanelWidth : this.$toolsPanel_.width();
 
-    let maxWidth = windowWidth - rightPanelWidth - minimumMapWidth;
+    let maxWidth = windowWidth - toolsPanelWidth - minimumMapWidth - this.appBarWidth_;
     if (maxWidth < this.dataPanelMinResizableWidth_) {
       maxWidth = this.dataPanelMinResizableWidth_;
     }
 
     this.$dataPanel_.resizable('option', 'maxWidth', maxWidth);
 
     const width = this.$dataPanel_.width();
-    if (!width) {
+    if (width !== 0 && !width) {
       throw new Error('Missing width');
     }
     if (width > maxWidth) {
       this.$dataPanel_.width(maxWidth);
+      this.dataPanelWidth_ = maxWidth;
+      this.map.updateSize();
+    }
+  }
+
+  /**
+   * Set the tools panel (on the right) maximum resizable width depending
+   * on the current size of the window, taking into consideration the
+   * width the left panel has.
+   *
+   * If, after resizing, the size of the data panel would be too big,
+   * resize it as well.
+   * @private
+   * @hidden
+   */
+  setToolsPanelMaxResizableWidth_(newDataPanelWidth) {
+    const minimumMapWidth = 120;
+    const windowWidth = window.innerWidth;
+
+    // this.$dataPanel_.width() often has not the new value yet after resizing, so take it from the resizing event!
+    const dataPanelWidth = newDataPanelWidth ? newDataPanelWidth : this.$dataPanel_.width();
+
+    let maxWidth = windowWidth - dataPanelWidth - minimumMapWidth - this.appBarWidth_;
+    if (maxWidth < this.toolsPanelMinResizableWidth_) {
+      maxWidth = this.toolsPanelMinResizableWidth_;
+    }
+
+    this.$toolsPanel_.resizable('option', 'maxWidth', maxWidth);
+
+    const width = this.$toolsPanel_.width();
+    if (width !== 0 && !width) {
+      throw new Error('Missing width');
+    }
+    if (width > maxWidth) {
+      this.$toolsPanel_.width(maxWidth);
```

Comment:
Same comments about duplicated lines.
And also for the dataPanelWidth and dataPanelWidth ?